packages/mui-material/src/TextField/TextField.js

```diff
@@ -37,7 +37,9 @@ const TextFieldRoot = styled(FormControl, {
   name: 'MuiTextField',
   slot: 'Root',
   overridesResolver: (props, styles) => styles.root,
-})({});
+})({
+  maxWidth: '100%',
```

Comment:
What is the reasoning behind adding this to `TextField` vs. `FormControl` vs. `InputBase`? I'm not fully familiar with this abstraction hierarchy.

Reply:
We want `maxWidth: 100%` to be applied directly to the `TextField` root because it's the component that's immediately impacted by the container's size. While you might consider applying it to `FormControl`, which `TextField` inherits, `FormControl` is primarily responsible for form-related context (like labels and helper text) and not for the direct sizing or spacing of `TextField`. Adding `maxWidth: 100%` to `FormControl` could create unintended side effects for other form controls.

As for `InputBase`, it's used by other components like `Select`. Applying `maxWidth: 100%` there would impact all components that rely on `InputBase`, potentially causing layout issues where this behavior isn't needed. And it won't even fix the issue because it is not a direct child of the container.

Reply:
Thanks for the explanation